azurerm/internal/services/sentinel/parse/sentinel_alert_rules.go

```diff
@@ -27,8 +33,48 @@ func SentinelAlertRuleID(input string) (*SentinelAlertRuleId, error) {
 		return nil, fmt.Errorf("parsing workspace part of Sentinel Alert Rule ID %q: %+v", input, err)
 	}
 	return &SentinelAlertRuleId{
+		Subscription:  workspaceId.Subscription,
 		ResourceGroup: workspaceId.ResourceGroup,
 		Workspace:     workspaceId.Name,
 		Name:          name,
 	}, nil
 }
+
+type SentinelAlertRuleActionId struct {
+	Subscription  string
+	ResourceGroup string
+	Workspace     string
+	Rule          string
+	Name          string
+}
+
+func (aid SentinelAlertRuleActionId) FormatSentinelAlertRuleId() SentinelAlertRuleId {
+	return SentinelAlertRuleId{
+		Subscription:  aid.Subscription,
+		ResourceGroup: aid.ResourceGroup,
+		Workspace:     aid.Workspace,
+		Name:          aid.Rule,
+	}
+}
+
+func SentinelAlertRuleActionID(input string) (*SentinelAlertRuleActionId, error) {
+	// Example ID: /subscriptions/<sub1>/resourceGroups/<grp1>/providers/Microsoft.OperationalInsights/workspaces/<workspace1>/providers/Microsoft.SecurityInsights/alertRules/<rule1>/actions/<action1>
+	groups := regexp.MustCompile(`^(.+)/actions/([^/]+)$`).FindStringSubmatch(input)
+	if len(groups) != 3 {
+		return nil, fmt.Errorf("faield to parse Sentinel Alert Rule Action ID: %q", input)
+	}
+
+	rule, name := groups[1], groups[2]
```

Comment:
rather than doing this, can we parse this and key into `alertRules` and `actions`?

Reply:
@tombuildsstuff I'm sorry I don't quite get your point here. Do you mean to embed `SentinelAlertRuleId` as a member inside `SentinelAlertRuleActionId`, and also embed `LogAnalyticsWorkspaceID` as a member inside `SentinelAlertRuleId`?

Reply:
sorry that was unclear - as in, we should be using the same ID Parsers as everything else here - to ensure another ID isn't being used inadvertantly - so can we make this:

```suggestion
	action := SentinelAlertRuleActionId{
		ResourceGroup: id.ResourceGroup,
	}
	if server.Name, err = id.PopSegment("actions"); err != nil {
		return nil, err
	}
	
	(etc)
```